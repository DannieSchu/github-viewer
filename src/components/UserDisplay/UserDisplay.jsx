import React from 'react';
import PropTypes from 'prop-types';

const UserDisplay = ({ avatar, fullName, username, followers, following, url }) => {
  return (
    <>
      <img src={avatar} alt={fullName} />
      <section>
        <article>
          <h2>{fullName}</h2>
          <h3>{username}</h3>
        </article>
        <article>
          <p>Followers:<br/>{followers}</p>
          <p>Following:<br/>{following}</p>
          <a href={url} target="blank">{url}</a>
        </article>
      </section>
    </>
  );
};

UserDisplay.propTypes = {
  avatar: PropTypes.string.isRequired,
  fullName: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  followers: PropTypes.string.isRequired,
  following: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default UserDisplay;
