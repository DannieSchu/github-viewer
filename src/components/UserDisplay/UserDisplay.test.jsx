import React from 'react';
import { shallow } from 'enzyme';
import UserDisplay from './UserDisplay.jsx';

describe('UserDisplay', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <UserDisplay 
        avatar="https://avatars0.githubusercontent.com/u/55268278?v=4"
        fullName="Dannie Schumaker"
        username="DannieSchu"
        followers="0"
        following="0"
        url="https://github.com/DannieSchu"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
