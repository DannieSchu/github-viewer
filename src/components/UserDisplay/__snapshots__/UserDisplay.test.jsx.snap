// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDisplay matches a snapshot 1`] = `
<Fragment>
  <img
    alt="Dannie Schumaker"
    src="https://avatars0.githubusercontent.com/u/55268278?v=4"
  />
  <section>
    <article>
      <h2>
        Dannie Schumaker
      </h2>
      <h3>
        DannieSchu
      </h3>
    </article>
    <article>
      <p>
        Followers:
        <br />
        0
      </p>
      <p>
        Following:
        <br />
        0
      </p>
      <a
        href="https://github.com/DannieSchu"
        target="blank"
      >
        https://github.com/DannieSchu
      </a>
    </article>
  </section>
</Fragment>
`;
